services/api/service.go

```diff
@@ -386,6 +405,67 @@ func (api *RelayAPI) startValidatorRegistrationDBProcessor() {
 	}
 }
 
+// simulateBlock sends a request for a block simulation to blockSimRateLimiter.
+func (api *RelayAPI) simulateBlock(opts blockSimOptions) error {
+	t := time.Now()
+	simErr := api.blockSimRateLimiter.send(opts.ctx, opts.req, opts.highPrio)
+	log := opts.log.WithFields(logrus.Fields{
+		"duration":   time.Since(t).Seconds(),
+		"numWaiting": api.blockSimRateLimiter.currentCounter(),
+	})
+	if simErr != nil {
+		log.WithError(simErr).Error("block validation failed")
+		return simErr
+	}
+	log.Info("block validation successful")
+	return nil
+}
+
+// Demotes a set of builders who have matching collateral IDs. This function
+// continues to try demoting even if an error is encountered for any builder.
+func (api *RelayAPI) demoteBuildersByCollateralID(builderPubkey string) {
```

Comment:
There's multiple ways this function can error out, and any of these might need to be manually followed up for executing the payment while the demotion not being part of the database. I suggest to at least label them commonly with a certain "relay-must-pay-but-db-failed"-like log tag.

Reply:
great point! in https://github.com/michaelneuder/mev-boost-relay/pull/2/commits/dd2fd606da6f27cee569fc3294882aeec43dac9a i added a logging field 

```
"errorWritingRefundToDB": true,
```

this should be seen as a critically important log!!